### PR TITLE
Commit team 5

### PR DESCRIPTION
**Pull request by DaniellaGeorgieva**
Даниела Георгиева
Таня Илиева
Калина Боева

### CODE REVIEW

File: Additional/Teams/Team№5/src/cpp/Shop.cpp [hunk 1]
```diff
@@ -0,0 +1,165 @@
+#include"Shop.h"
+#include"Casual.h"
+#include"Couture.h"
+#include"SportsWear.h"
+
+Shop::Shop()
+{
+	shopName = "";
+	address = "";
```
Comment (nikola354) [inline Additional/Teams/Team№5/src/cpp/Shop.cpp:9]: you can remove these lines. the default constructor of String makes the same thing

File: Additional/Teams/Team№5/src/h/Shop.h [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+private:
+	String shopName;
+	String address;
+	double profit;
+	ClothingContainer listOfClothes;
+public:
+	Shop();
+
+	String getName() const;
+	String getAddress() const;
```
Comment (nikola354) [inline Additional/Teams/Team№5/src/h/Shop.h:16]: const String& ?

Comment (nikola354) [inline Additional/Teams/Team№5/src/h/Shop.h:16]: const String& ?

File: Additional/Teams/Team№5/src/h/Shop.h [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	double getProfit() const;
+
+	void setName(const String&);
+	void setAddress(const String&);
+	void addToProfit(const double);
+
+	void displayMenu();
+	
+private:
+	void diplayAddDialog();
```
Comment (nikola354) [inline Additional/Teams/Team№5/src/h/Shop.h:26]: typo

File: Additional/Teams/Team№5/src/cpp/Shop.cpp [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+	String size;
+	std::cin >> size;
+	std::cout << "Type (C for Casual, S for SportsWear, D for Couture/Designer): \n>";
+	char type;
+	std::cin >> type;
+	std::cout << "Price: \n>";
+	double price;
+	std::cin >> price;
+	std::cout << "Count to add (number > 0): \n>";
+	size_t countToAdd;
```
Comment (nikola354) [inline Additional/Teams/Team№5/src/cpp/Shop.cpp:63]: check if the price is positive number

File: Additional/Teams/Team№5/src/h/ClothingContainer.h [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+	size_t capacity;
+	size_t size;
+	size_t logSize;
+public:
+	ClothingContainer();
+	ClothingContainer(const ClothingContainer&);
+	ClothingContainer& operator=(const ClothingContainer&);
+	~ClothingContainer();
+
+	double getPrice(size_t);
```
Comment (GeorgiTerziev02) [inline Additional/Teams/Team№5/src/h/ClothingContainer.h:19]: const function

File: Additional/Teams/Team№5/src/h/ClothingContainer.h [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+	void AddClothingItem(ClothingItem&);
+	bool DeleteClothingItem(size_t);
+	void ListAvailableItems();
+	void ListSoldItems();
```
Comment (GeorgiTerziev02) [inline Additional/Teams/Team№5/src/h/ClothingContainer.h:23]: List functions should be const

File: Additional/Teams/Team№5/src/h/ClothingContainer.h [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+	void ExportSoldLog();
```
Comment (GeorgiTerziev02) [inline Additional/Teams/Team№5/src/h/ClothingContainer.h:24]: again const function

File: Additional/Teams/Team№5/src/cpp/ClothingContainer.cpp [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+	{
+		writeFile << log[i] << "\nPrice: " << logPrice[i] << std::endl;
+	}
+}
+
+void ClothingContainer::resize()
+{
+	ClothingItem** tempCont = new ClothingItem * [capacity * 2];
+	String* tempLog = new String[capacity * 2];
+	double* tempLogPrice = new double[capacity * 2];
```
Comment (GeorgiTerziev02) [inline Additional/Teams/Team№5/src/cpp/ClothingContainer.cpp:117]: Why are you resizing everything in one method?
Example: Line 69
You are removing an item, but you call resize later so you are resizing the container and the log, when you only needed to resize the log

File: Additional/Teams/Team№5/src/h/ClothingContainer.h (hunk 7 above)
Comment (GeorgiTerziev02) [inline Additional/Teams/Team№5/src/h/ClothingContainer.h:24]: The above 3 methods should be const

File: Additional/Teams/Team№5/src/cpp/Shop.cpp [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+			diplayAddDialog();
+		}
+		else if (cmd == "sell" || cmd == "Sell") {
+			displaySellDialog();//delete function
+		}
+		else if (cmd == "list" || cmd == "List") {
+			displayListDialog();
+		}
+		else if (cmd == "export" || cmd == "Export") {
+			//function from container
```
Comment (GeorgiTerziev02) [inline Additional/Teams/Team№5/src/cpp/Shop.cpp:156]: //do something or delete the if

**Review by GeorgiTerziev02 — CHANGES_REQUESTED** [review]

File: Additional/Teams/Team№5/src/h/ClothingContainer.h (hunk 6 above)
Comment (GeorgiTerziev02) [reply Additional/Teams/Team№5/src/h/ClothingContainer.h:23]: + function names should be camelCase - starting with lower letter

**Review by GeorgiTerziev02 — CHANGES_REQUESTED** [review]
Looks fine but it needs some minor fixes

**Review by nikola354 — APPROVED** [review]

**Review by GeorgiTerziev02 — APPROVED** [review]

**Review by Justsvetoslavov — COMMENTED** [review]
Diagram is incorrect.

Clothing shop has a "Clothing Container" (composition) and clothing container consists of cloths(clothingItem) , again composition.

If you are struggling to understand this, ping me to discuss it.

File: Additional/Teams/Team№5/src/cpp/ClothingItem.cpp [hunk 10]
```diff
@@ -0,0 +1,151 @@
+#include"ClothingItem.h"
+
+ClothingItem::ClothingItem() : name(), price(0), numAvailable(0)
+{}
```
Comment (Justsvetoslavov) [inline Additional/Teams/Team№5/src/cpp/ClothingItem.cpp:4]: move to line 3.

File: Additional/Teams/Team№5/src/cpp/ClothingItem.cpp [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+ClothingItem::ClothingItem(const String& name, const double price, const size_t numAvailable) : name(name), price(price), numAvailable(numAvailable)
+{}
```
Comment (Justsvetoslavov) [inline Additional/Teams/Team№5/src/cpp/ClothingItem.cpp:6]: move to line 5

File: Additional/Teams/Team№5/src/cpp/ClothingItem.cpp [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+	this->numAvailable = numAvailable;
+}
+void ClothingItem::setGender(const char& gender)
+{
+	if (gender == 'M' || gender == 'm')
+	{
+		this->gender = Gender::MAN;
+		return;
+	}
+
```
Comment (Justsvetoslavov) [inline Additional/Teams/Team№5/src/cpp/ClothingItem.cpp:28]: remove empty lines.

File: Additional/Teams/Team№5/src/cpp/ClothingItem.cpp [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+void ClothingItem::setNumAvailable(const size_t numAvailable)
+{
+	this->numAvailable = numAvailable;
+}
+void ClothingItem::setGender(const char& gender)
+{
+	if (gender == 'M' || gender == 'm')
+	{
+		this->gender = Gender::MAN;
+		return;
```
Comment (Justsvetoslavov) [inline Additional/Teams/Team№5/src/cpp/ClothingItem.cpp:26]: returns are unnecessary ,as well as brackets, statements will be 1 liners.

File: Additional/Teams/Team№5/src/cpp/ClothingItem.cpp [hunk 14]
```diff
@@ -0,0 +1,151 @@
...
+
+	else if (gender == 'U' || gender == 'u')
+	{
+		this->gender = Gender::UNISEX;
+		return;
+	}
+}
+void ClothingItem::setSize(const String& size)
+{
+	if (size == "XS" || size == "xs")
```
Comment (Justsvetoslavov) [inline Additional/Teams/Team№5/src/cpp/ClothingItem.cpp:43]: remove brackets , 1 liners.

File: Additional/Teams/Team№5/src/cpp/ClothingItem.cpp [hunk 15]
```diff
@@ -0,0 +1,151 @@
...
+		return "S";
+	case ClothingItem::Size::M:
+		return "M";
+	case ClothingItem::Size::L:
+		return "L";
+	case ClothingItem::Size::XL:
+		return "XL";
+	case ClothingItem::Size::XXL:
+		return "XXL";
+	case ClothingItem::Size::UNKNOWN:
```
Comment (Justsvetoslavov) [inline Additional/Teams/Team№5/src/cpp/ClothingItem.cpp:128]: could be considered as the default.

File: Additional/Teams/Team№5/src/cpp/Couture.cpp [hunk 16]
```diff
@@ -0,0 +1,30 @@
+#include "Couture.h"
+
+Couture::Couture(const String& name, const double price, const size_t numAvailable) :ClothingItem(name, price, numAvailable)
+{}
```
Comment (Justsvetoslavov) [inline Additional/Teams/Team№5/src/cpp/Couture.cpp:4]: move to line 3

File: Additional/Teams/Team№5/src/cpp/ClothingContainer.cpp [hunk 17]
```diff
@@ -0,0 +1,177 @@
...
+	{
+		log[i] = other.log[i];
+		logPrice[i] = other.logPrice[i];
+	}
+}
+
+void ClothingContainer::free()
+{
+	for (size_t i = 0; i < size; i++)
+	{
```
Comment (Justsvetoslavov) [inline Additional/Teams/Team№5/src/cpp/ClothingContainer.cpp:170]: remove brackets.

File: Additional/Teams/Team№5/src/cpp/ClothingContainer.cpp [hunk 18]
```diff
@@ -0,0 +1,177 @@
...
+	capacityContainer = other.capacityContainer;
+	capacityLog = other.capacityLog;
+	size = other.size;
+	logSize = other.logSize;
+	container = new ClothingItem * [capacityContainer];
+	log = new String[capacityLog];
+	logPrice = new double[capacityLog];
+
+	for (size_t i = 0; i < size; i++)
+	{
```
Comment (Justsvetoslavov) [inline Additional/Teams/Team№5/src/cpp/ClothingContainer.cpp:156]: remove brackets.

File: Additional/Teams/Team№5/src/cpp/ClothingContainer.cpp [hunk 19]
```diff
@@ -0,0 +1,177 @@
...
+		writeFile << log[i] << "\nPrice: " << logPrice[i] << std::endl;
+	}
+}
+
+void ClothingContainer::resizeContainer()
+{
+	ClothingItem** tempCont = new ClothingItem * [capacityContainer *= 2];
+	for (size_t i = 0; i < size; i++)
+	{
```
Comment (Justsvetoslavov) [inline Additional/Teams/Team№5/src/cpp/ClothingContainer.cpp:118]: remove brackets.

File: Additional/Teams/Team№5/src/cpp/ClothingContainer.cpp [hunk 20]
```diff
@@ -0,0 +1,177 @@
...
+		std::cout << log[i] << "\nPrice: " << logPrice[i] << std::endl;
+	}
+}
+
+void ClothingContainer::exportSoldLog() const
+{
+	std::ofstream writeFile("logFile.txt", std::ios::app);
+	for (size_t i = 0; i < logSize; i++)
+	{
```
Comment (Justsvetoslavov) [inline Additional/Teams/Team№5/src/cpp/ClothingContainer.cpp:109]: remove brackets.

File: Additional/Teams/Team№5/src/cpp/ClothingContainer.cpp [hunk 21]
```diff
@@ -0,0 +1,177 @@
...
+	container[size - 1] = nullptr;
+	size--;
+
+	return true;
+}
+
+void ClothingContainer::listAvailableItems() const
+{
+	for (size_t i = 0; i < size; i++)
+	{
```
Comment (Justsvetoslavov) [inline Additional/Teams/Team№5/src/cpp/ClothingContainer.cpp:92]: remove brackets.

File: Additional/Teams/Team№5/src/cpp/ClothingContainer.cpp [hunk 22]
```diff
@@ -0,0 +1,177 @@
...
+	}
+	if (size >= capacityContainer)
+	{
+		resizeContainer();
+	}
+
+	container[size++] = item.clone();
+}
+
+bool ClothingContainer::deleteClothingItem(size_t index) const
```
Comment (Justsvetoslavov) [inline Additional/Teams/Team№5/src/cpp/ClothingContainer.cpp:61]: adjust 1 liners.

**Review by Justsvetoslavov — CHANGES_REQUESTED** [review]

**Pull request merged**